src/shell/bash.rs

```diff
@@ -16,7 +16,23 @@ impl Shell for Bash {
         let path = path
             .to_str()
             .ok_or_else(|| anyhow::anyhow!("Can't convert path to string"))?;
-        Ok(format!("export PATH={:?}:$PATH", path))
+        let multishell_storage = crate::directories::multishell_storage();
+        Ok(formatdoc!(
+            r#"
+                new_path="{path}"
+                IFS=":"
+                for p in $PATH; do
+                    if ! echo $p | grep -q {multishell_storage}; then
+                        new_path="$new_path:$p"
+                    fi
+                done
+                unset IFS
+                export PATH=$new_path
+                unset new_path
```

Comment:
hmmmm. what if a user had a `IFS` variable prior to `eval "$(fnm env)"`? it would probably be unset, right? (I'm not sure, but I assume it shares the same context)

Can we make these parts written in Rust instead of bash/other shell scripting language?
I want to have the minimum amount of shell code in fnm

Reply:
Yes, we can, but we have to manage `split_paths()` and `join_paths()` manually. Due to different types of path separators between OS.

> hmmmm. what if a user had a IFS variable prior to eval "$(fnm env)"? it would probably be unset, right? (I'm not sure, but I assume it shares the same context)

`IFS` already has a default value. However, after we unset it, Bash will process word extraction using its default value.